src/nsrt_learning.py

```diff
@@ -36,134 +61,161 @@ def learn_nsrts_from_data(dataset: Dataset, predicates: Set[Predicate],
-                   trans_delete_effects for o in atom.objects}
-        objects.update(option.objects)
+
+    # Prune partitions with not enough data.
+    kept_idxs = []
+    for idx, partition in enumerate(partitions):
+        if len(partition) >= CFG.min_data_for_nsrt:
+            kept_idxs.append(idx)
+    params = [params[i] for i in kept_idxs]
+    add_effects = [add_effects[i] for i in kept_idxs]
+    delete_effects = [delete_effects[i] for i in kept_idxs]
+    partitions = [partitions[i] for i in kept_idxs]
+
+    # Learn preconditions.
+    preconds = [_learn_preconditions(p) for p in partitions]
+
+    # Finalize the operators.
+    ops = []
+    for i in range(len(params)):
+        name = f"Operator{i}"
+        op = STRIPSOperator(name, params[i], preconds[i], add_effects[i],
+                            delete_effects[i])
+        print("Learned STRIPSOperator:")
+        print(op)
+        ops.append(op)
+
+    return ops, partitions
+
+
+def learn_options(
+    strips_ops: List[STRIPSOperator],
+    partitions: List[Partition],
+    ) -> List[Tuple[ParameterizedOption, List[Variable]]]:
+    """Learn options for segments, or just look them up if they're given.
+    """
+    assert not CFG.do_option_learning, "TODO: implement option learning."
+    del strips_ops  # unused
+    return _extract_options_from_data(partitions)
+
+
+def _extract_options_from_data(
+    partitions: List[Partition],
+    ) -> List[Tuple[ParameterizedOption, List[Variable]]]:
+    """Look up the options from the data.
+    """
+    option_specs = []
+    for partition in partitions:
+        for i, (segment, sub) in enumerate(partition):
+            option = segment.actions[0].get_option()
+            if i == 0:
+                param_option = option.parent
+                option_vars = [sub[o] for o in option.objects]
+            else:
+                assert param_option == option.parent
+                assert option_vars == [sub[o] for o in option.objects]
+            # Make sure the option is consistent within a trajectory.
+            for a in segment.actions:
+                option_a = a.get_option()
+                assert param_option == option_a.parent
+                assert option_vars == [sub[o] for o in option_a.objects]
+        option_specs.append((param_option, option_vars))
+    return option_specs
+
+
+def _find_option_for_segment(segment: Segment,
+                             param_option: ParameterizedOption,
+                             opt_vars: Sequence[Variable]) -> None:
+    """Figure out which option was executed within the segment.
+
+    At this point, we know which ParameterizedOption was used in the segment,
+    and we know the option_vars, but we don't know what parameters were used.
+
+    Modifies segment in place.
+    """
+    assert not CFG.do_option_learning, "TODO: implement option learning."
+    assert not segment.has_option()
+    segment.set_option_from_trajectory()
```

Comment:
I'm confused, when did the trajectories have options added to them? (In the option learning code path)

Reply:
In the non-option learning code path, couldn't we just do this in extract_options_from_data?